File: backend/src/main/java/com/traveloop/service/interfaces/TripService.java
package com.traveloop.service.interfaces;

import com.traveloop.model.dto.request.TripRequest;
import com.traveloop.model.dto.response.TripResponse;

import java.util.List;
import java.util.UUID;

public interface TripService {

    TripResponse createTrip(UUID userId, TripRequest request);

    TripResponse getTripById(UUID tripId, UUID userId);

    TripResponse getTripByShareToken(String shareToken);

    List<TripResponse> getUserTrips(UUID userId);

    List<TripResponse> getSharedTrips(UUID userId);

    List<TripResponse> getPublicTrips();

    TripResponse updateTrip(UUID tripId, UUID userId, TripRequest request);

    TripResponse updateTripStatus(UUID tripId, UUID userId, String status);

    String generateShareToken(UUID tripId, UUID userId);

    void deleteTrip(UUID tripId, UUID userId);

    void addCollaborator(UUID tripId, UUID ownerId, String collaboratorEmail, String role);

    void removeCollaborator(UUID tripId, UUID ownerId, UUID collaboratorId);
}

File: frontend/src/services/tripService.js
/* ────────────────────────────────────────────
   tripService — Trip CRUD API calls
   ──────────────────────────────────────────── */

import api from './api';

const tripService = {
  createTrip: (data) => api.post('/api/trips', data).then((r) => r.data),

  getMyTrips: () => api.get('/api/trips/my').then((r) => r.data),

  getTripById: (tripId) => api.get(`/api/trips/${tripId}`).then((r) => r.data),

  updateTrip: (tripId, data) =>
    api.put(`/api/trips/${tripId}`, data).then((r) => r.data),

  deleteTrip: (tripId) => api.delete(`/api/trips/${tripId}`),

  updateTripStatus: (tripId, status) =>
    api.patch(`/api/trips/${tripId}/status`, { status }).then((r) => r.data),

  generateShareLink: (tripId) =>
    api.post(`/api/trips/${tripId}/share`).then((r) => r.data),

  getPublicTrips: () => api.get('/api/trips/public').then((r) => r.data),

  getTripByShareToken: (token) =>
    api.get(`/api/trips/share/${token}`).then((r) => r.data),
};

export default tripService;
